File: backend/app/schemas/judge.py
from pydantic import BaseModel


class JudgeRequest(BaseModel):
    odai: str
    answer: str


class JudgeScore(BaseModel):
    judge: str
    score: int
    comment: str


class JudgeResult(BaseModel):
    id: str
    odai: str
    answer: str
    total_score: int
    judges: list[JudgeScore]


class JudgeResponse(BaseModel):
    id: str
    total_score: int
    judges: list[JudgeScore]

File: frontend/src/types/judge.ts
export type JudgeScore = {
  judge: string;
  score: number;
  comment: string;
};

export type JudgeResponse = {
  id: string;
  total_score: number;
  judges: JudgeScore[];
};

export type JudgeResult = {
  id: string;
  odai: string;
  answer: string;
  total_score: number;
  judges: JudgeScore[];
};
